### ISSUE TITLE
httpslib.HTTPSConnection is broken on py3

### ISSUE THREAD

**Issue by novocaine** [M2Crypto/M2Crypto#26]
(I've been trying to make the python3 branch work on windows and found this along the way while trying to run the tests against my win32 patch).

ssl.Connection is broken on python 3. This class attempts to emulate being a socket just enough that it can work with the standard python http.client.HTTPConnection, and doesn't quite make it on python 3.

*

**Comment by mcepl** [comment]
> Preference?

No, preference. I am going to send you the email with more details of my current position on M2Crypto.
